### PR TITLE
video_core: implement PM4 disassembler

### PR DESCRIPTION
**Pull request by Random06457**
This PR adds a simple PM4 disassembler for debugging purpose.

The disassembled PM4 display lists are saved to disk next to their binary version when `Config::dumpPM4()` is true.

Not all PM4 commands are supported yet but the ones supported by the emulator should disassemble fine.

### CODE REVIEW

**Comment by psucien** [comment]
Thank you for your efforts, having a pm4 analyzer is indeed very helpful. At this moment we already have a tool for this purpose, and probably it will be good if you check it out: https://github.com/psucien/tlg-emu-tools
You can reach anyone from dev team in Discord if you are interested in its development.

File: src/core/libraries/gnmdriver/gnmdriver.cpp [hunk 1]
```diff
@@ -320,6 +323,296 @@ static void WaitGpuIdle() {
...
+        const auto* header = reinterpret_cast<const PM4Header*>(cmds.data());
+        const u32 type = header->type;
+        switch (type) {
+        case 0:
+            append(fmt::format("PM4Type0: base={}, count={}\n", header->type0.base.Value(),
+                               header->type0.count.Value()));
+            for (size_t i = 0; i < header->type0.count.Value(); i++)
+                append(fmt::format("- 0x{:08X}\n", cmds.data()[1 + i]));
+            break;
+            cmds = cmds.subspan(header->type0.NumWords() + 1);
```
Comment (baggins183) [inline src/core/libraries/gnmdriver/gnmdriver.cpp:351]: break should be after this line

Comment (Random06457) [reply src/core/libraries/gnmdriver/gnmdriver.cpp:351]: Thanks

**Comment by Hermiten** [comment]
What do you plan to do with this @psucien ?

**Comment by psucien** [comment]
Unfortunatelly it won't go by at least three reasons:
- high chance of sync issues caused by heavy io in submission functions. These functions shouldn't be blocking and we already risking here by having binary pm4 dump;
- some games show about a megabyte binary dumps _per frame_. Imagine this being written as text and accumulated on disk for a some time;
- since we aready have ImGui integrated, we are going to move the existing parser as overlay into main emulator window.

**Comment by psucien** [comment]
Done in scope of https://github.com/shadps4-emu/shadPS4/pull/1094

**Pull request closed without merging**